openwisp_radius/receivers.py

```diff
@@ -32,20 +32,23 @@ def send_email_on_new_accounting_handler(sender, accounting_data, view, **kwargs
 
 def set_default_group_handler(sender, instance, created, **kwargs):
     if created:
-        RadiusGroup = load_model('RadiusGroup')
-        RadiusUserGroup = load_model('RadiusUserGroup')
-        queryset = RadiusGroup.objects.filter(
-            default=True, organization_id=instance.organization_id
-        )
-        if (
-            queryset.exists()
-            and not instance.user.radiususergroup_set.filter(
-                group__organization_id=instance.organization_id
-            ).exists()
-        ):
-            ug = RadiusUserGroup(user=instance.user, group=queryset.first())
-            ug.full_clean()
-            ug.save()
+        try:
+            RadiusGroup = load_model('RadiusGroup')
+            RadiusUserGroup = load_model('RadiusUserGroup')
+            queryset = RadiusGroup.objects.filter(
+                default=True, organization_id=instance.organization_id
+            )
+            if (
+                queryset.exists()
+                and not instance.user.radiususergroup_set.filter(
+                    group__organization_id=instance.organization_id
+                ).exists()
+            ):
+                ug = RadiusUserGroup(user=instance.user, group=queryset.first())
+                ug.full_clean()
+                ug.save()
+        except Exception as e:
```

Comment:
PS: a failing test is missing

Reply:
> We usually wrap in the try/except only the lines of code which we expect to fail and we must be specific in what exception we expect. This code can create more problems than what it's trying to solve because it can potentially hide other unwanted behavior. On operations like these which are not critical, letting them failing loudly is better than failing silently.

Sure, I think it would be better to wrap the try-except block inside the second nested if-statement instead, that being the part which prevents adding organization group and assigning radius group together. Is that a fair proposition?

Reply:
@pandafy thoughts?

Reply:
@Shiva953 that is correct. 

Another thing, we don't want to catch all types of exceptions here. We shall check the exception class of the error raised (probably integrity error) and add handling only for that.